Visualizacion del Restaurante

EnriDv opened the pull request:
Se creo la visualizacion correcta del restaurante a traves de su ID. 
Se cambio el nombre de Restaurante a Business para mantener la coherencia con los demas archivos. 
Se creo el componente ProductList para cargar la lista de productos del restaurante y cargarlos desde la base de datos.
Se creo el componente ProductCard para visualizar cada producto de la lista de productos.
Se actualizo la ruta en el App

EnriDv committed:
Visualizacion del Restaurante

Se creo la visualizacion correcta del restaurante a traves de su ID.
Se cambio el nombre de Restaurante a Business para mantener la coherencia con los demas archivos.
Se creo el componente ProductList para cargar la lista de productos del restaurante y cargarlos desde la base de datos
Se creo el componente ProductCard para visualizar cada producto de la lista de productos.
Se actualizo la ruta en el App

EnriDv commented:
@fcapobiancot @RaidenMT

EnriDv committed:
Update App.jsx

Se corrigio la etiqueta de Restaurante por Business

fcapobiancot commented on OrderNow-React/src/components/ProductCard.jsx:
Eliminar los comentarios

fcapobiancot commented on OrderNow-React/src/components/ProductList.jsx:
Eliminar los comentarios

fcapobiancot commented on OrderNow-React/src/pages/Business.jsx:
```diff
@@ -0,0 +1,65 @@
+        .eq('id', id)
+        .single();
+
+      if (error) {
+        console.error("Error al obtener el negocio:", error);
+      } else {
+        setBusiness(data);
+      }
+      setLoading(false);
+    };
```
Eliminar los console outputs ademas de manejar los errores en el bloque if y lo demas sacarlo del else para mejorar legibilidad y buenas practicas

EnriDv replied:
El console Output me sirve para manejar mejor los errores, y la verdad no se como mas manejarlos :p

RaidenMT replied:
por ahora dejemos los console.error, pero también agreguemos los alert(). No olvidarse que también menciona Fernando quitar el else.

RaidenMT replied:
Falta quitar las llaves del else para dejar libre setBusiness. Sabrías hacer el código @EnriDv ?

EnriDv committed:
Agregado estado de Cerrado a los Business

Se agrego la logica para manejar el cierre del restaurante.
Al cerrar el restaurante, el texto pierde opacidad y se agrega una advertencia de que el restaurante esta cerrado.
El restaurante cierra cuando no esta en horas activas.
El restaurante cierra cuando el estado es is_open es Falso.

fcapobiancot commented on OrderNow-React/src/pages/Business.jsx:
```diff
@@ -28,34 +28,56 @@ function Business() {
+        }`}
+      >
         <div className="flex items-center gap-6">
+          {/* Si cuentas con imagen, reemplaza este div */}
```
elimnar comentario

fcapobiancot commented on OrderNow-React/src/pages/Business.jsx:
```diff
@@ -28,34 +28,56 @@ function Business() {
   return (
     <main className="max-w-6xl mx-auto px-4 pt-24 pb-8">
-      <div className="bg-white rounded-lg shadow-md p-6 mb-8">
+      {/* Se puede cambiar por un componente de advertencia */}
```
eliminar comentario

EnriDv replied:
El comentario es para poder guiarme si me toca la tarea de hacer los mensajes de advertencia, asi saber donde puedo llamarlos

RaidenMT replied:
Entonces documentarlos como 

{/* TODO: Se puede cambiar por un componente de advertencia */}

fcapobiancot commented on OrderNow-React/src/pages/Business.jsx:
```diff
@@ -18,7 +18,7 @@ function Business() {
+        console.error('Error al obtener el negocio:', error.message);
       } else {
         setBusiness(data);
       }
```
manejar los errores en el bloque if y retornar. Lo demas sacarlo del else.

EnriDv replied:
Como ya dije, el console Output me sirve para manejar mejor los errores como son de la base de datos, y la verdad no se como mas manejarlos :p

EnriDv commented:
@RaidenMT

RaidenMT commented on OrderNow-React/src/pages/Business.jsx:
```diff
@@ -0,0 +1,65 @@
+    fetchBusiness();
+  }, [id, supaBaseCom]);
+
+  if (loading) {
```
¿Esto funciona correctamente? Solo sale esto o muestra el return de abajo? de la lína 39?

EnriDv replied:
si funciona, se muestra el mensaje por poco tiempo hasta que termina de obtener los datos
![image](https://github.com/user-attachments/assets/fcf655f8-a3af-45c7-9a83-722ef49313db)

RaidenMT commented on OrderNow-React/src/pages/Business.jsx:
```diff
@@ -0,0 +1,65 @@
+    return <p>Cargando detalles del restaurante...</p>;
+  }
+
+  if (!business) {
```
La misma pregunta para la de arriba

EnriDv replied:
Igualmente, cuando no se encuentra el restaurante (id) se muestra unicamente el mensaje.
![image](https://github.com/user-attachments/assets/b46e5ca1-b8cc-46d8-b596-b2893f0982a3)

RaidenMT commented on OrderNow-React/src/components/ProductList.jsx:
```diff
@@ -0,0 +1,47 @@
+import React, { useState, useEffect } from 'react';
+import getSupaBaseClient from '../supabase-client'; // Asegúrate de la ruta correcta
```
Eliminar comentario

EnriDv replied:
El comentario se elimino en el siguiente commit 👌

RaidenMT reviewed (changes requested):
revisar por favor

EnriDv committed:
Correcciones Aplicadas

Se quito el if else en la seccion de error, ahora se muestra una alerta de que no se pudieron obtener los datos
Tambien se modifico el comentario de la advertencia

fcapobiancot reviewed (approved):
👍

RaidenMT reviewed (changes requested):
Revisar último comentario y hacer git PULL
